cep-9999.md

```diff
@@ -0,0 +1,380 @@
+          "file_extensions": None, # file extensions to associate with shortcut in registry
+          "url_protocols": None, # URI protocols to associate with shorcut in registry
+          "app_user_model_id": None, # Identifier used to associate processes with a taskbar icon
+        }
+      }
+    }
+  ]
+}
+```
+
+Note that each `platforms` sub-dictionary (`linux`, `macos`, `win`) can override the global values
+of their `menu_items[*]` entry (e.g. redefining `command` to adjust the shell syntax).
+
+Each JSON file MUST be validated against its `$id` schema at build time; e.g in `conda-build`.
+
+## Placeholders
+
+Each platform MUST provide these placeholders, to be used in the value of any `str`-accepting key.
+To be replaced, they MUST be wrapped in double curly braces: `{{ NAME }}`.
+
+Placeholder | Value
+------------|-------
+`BASE_PREFIX` | Path to the base Python location. In `conda` terms, this is the `base` environment
+`DISTRIBUTION_NAME` | Name of the base prefix directory; e.g. if `BASE_PREFIX` is `/opt/my-project`, this is `my-project`.
+`PREFIX` | Path to the target Python location. In `conda` terms, this is the path to the environment that contains the JSON file for this menu item. In some cases, it might be the same as `BASE_PREFIX`.
+`ENV_NAME` | Same as `DISTRIBUTION_NAME`, but for `PREFIX`.
+`PYTHON` | Path to the `python` executable in `PREFIX`.
+`BASE_PYTHON` | Path to the `python` executable in `BASE_PREFIX`.
+`MENU_DIR` | Path to the `Menu` directory in `PREFIX`.
+`MENU_ITEM_LOCATION` | Path to the main menu item artifact once installed. On Linux, this is the path to the `.desktop` file, on macOS it is the path to the `.app` directory, and on Windows it is the path to the Start Menu `.lnk` file.
+`BIN_DIR` | Path to the `bin` (Unix) or `Library/bin` (Windows) directories in `PREFIX`.
+`PY_VER` | Python `major.minor` version in `PREFIX`.
+`SP_DIR` | Path to Python's `site-packages` directory in `PREFIX`.
+`HOME` | Path to the user directory (`~`).
+`ICON_EXT` | Extension of the icon file expected by each platform. `png` in Linux, `icns` in macOS, and `ico` in Windows. Note the dot is _not_ included.
+**macOS-only** |
+`PYTHONAPP` | Path to the `python` executable installed in `PREFIX`, assuming the `python.app` conda package is installed. Equivalent to `{{ PREFIX }}/python.app/Contents/MacOS/python`. 
+**Windows-only** | 
+`SCRIPTS_DIR` | Path to the `Scripts` directory in `PREFIX`. 
+`BASE_PYTHONW` | Path to the `pythonw.exe` executable in `BASE_PREFIX`.
+`PYTHONW` | Path to the `pythonw.exe` executable in `PREFIX`.
+
+## Packaging guidelines
+
+`conda` packages that wish to create a shortcut at install time MUST provide a JSON file such that:
+
+- The JSON contents MUST pass schema validation.
+- The JSON file MUST be placed under `$PREFIX/Menu`.
+- The JSON file MUST include the package name in its filename. We recommend sticking to
+  `<pkg-name>.json` for simplicity.
```

Comment:
Not going to make this a blocker, but is there a reason not to make this `MUST BE <pkg-name>.json`? Having to implement some regex or `*${pkg-name}*.json` glob seems unnecessarily complicated and error prone; e.g., what if `pkg-name` is a substring of some other package?

Reply:
Current logic in `conda` [doesn't check the filename](https://github.com/conda/conda/blob/1b53c20dab60a262a87d21ed3609430117ff7e18/conda/core/path_actions.py#L569-L577). This is only used for filtering which packages should be processed from menuinst, which indeed [does check strictly for a full match](https://github.com/conda/constructor/blob/b6de413b2c40bf3caf7a809a1fbb08a8f1ee03a3/constructor/nsis/_nsis.py#L122-L136).

So you are right, this is better encoded as your suggestion. I'll fix it.